File: AdvantageFramework/Database/Views/AlertView.vb
Namespace Database.Views

    <Table("WV_MYALERTSLIST")>
    Public Class AlertView
        Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

        Public Enum Properties
            AlertID
            Subject
            Generated
            Processed
            UserName
            EmployeeCode
            Priority
            Type
            Category
            AlertTypeID
            AlertCategoryID
            AttachmentCount
            AlertLevel
            OfficeCode
            ClientCode
            DivisionCode
            ProductCode
            CampaignCode
            JobNumber
            JobComponentNumber
            SenderEmployeeCode
            DismissedDate
            ClientName
            DivisionName
            ProductDescription
            CampaignName
            Body
            LowerSubject
            LowerBody
            OfficeName
            JobDescription
            JobComponentDescription
            AlertUser
            ClientPortalAlert
            NewAlert
            ReadAlert
            VendorCode
            DueDate
            TimeDue
            AlertStateID
            AlertStateName
            CurrentNotify
            CurrentNotifyEmployeeCode
            CurrentNotifyEmployeeFullName
            GroupOffice
            GroupClient
            GroupDivision
            GroupProduct
            GroupJob
            GroupJobComponent
            TaskSequenceNumber
            TaskFunctionCode
            TaskFunctionDescription
            EstimateNumber
            EstimateComponentNumber
            GroupEstimateNumber
            GroupEstimateComponent
            GroupTask
            GroupCampaign
            CampaignID
            Version
            Build
            AlertSequenceNumber
            GroupPriority
            GroupDueDate
            GroupDueDateDisplay
            ID
            Sent
            AlertNotifyHeaderID
            AlertNotifyName
            VendorName

        End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

        <Key>
        <Required>
        <Column("ALERT_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertID() As Integer
        <MaxLength(254)>
        <Column("SUBJECT", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Subject() As String
        <Column("GENERATED")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Generated() As Nullable(Of Date)
        <Column("PROCESSED")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Processed() As Nullable(Of Date)
        <MaxLength(64)>
        <Column("USER_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property UserName() As String
        <MaxLength(6)>
        <Column("EMP_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property EmployeeCode() As String
        <Required>
        <Column("PRIORITY")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Priority() As Integer
        <Required>
        <MaxLength(40)>
        <Column("TYPE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Type() As String
        <Required>
        <MaxLength(40)>
        <Column("CATEGORY", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Category() As String
        <Required>
        <Column("ALERT_TYPE_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertTypeID() As Integer
        <Required>
        <Column("ALERT_CAT_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertCategoryID() As Integer
        <Column("ATTACHMENTCOUNT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AttachmentCount() As Nullable(Of Integer)
        <MaxLength(50)>
        <Column("ALERT_LEVEL", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertLevel() As String
        <MaxLength(4)>
        <Column("OFFICE_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property OfficeCode() As String
        <MaxLength(6)>
        <Column("CL_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ClientCode() As String
        <MaxLength(6)>
        <Column("DIV_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property DivisionCode() As String
        <MaxLength(6)>
        <Column("PRD_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ProductCode() As String
        <MaxLength(6)>
        <Column("CMP_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property CampaignCode() As String
        <Column("JOB_NUMBER")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property JobNumber() As Nullable(Of Integer)
        <Column("JOB_COMPONENT_NBR")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property JobComponentNumber() As Nullable(Of Short)
        <MaxLength(6)>
        <Column("SENDER_EMP_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property SenderEmployeeCode() As String
        <MaxLength(10)>
        <Column("DISMISSED_DATE")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property DismissedDate() As String
        <MaxLength(40)>
        <Column("CL_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ClientName() As String
        <MaxLength(40)>
        <Column("DIV_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property DivisionName() As String
        <MaxLength(40)>
        <Column("PRD_DESCRIPTION", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ProductDescription() As String
        <MaxLength(128)>
        <Column("CMP_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property CampaignName() As String
        <Column("BODY")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Body() As String
        <MaxLength(254)>
        <Column("LOWER_SUBJECT", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property LowerSubject() As String
        <MaxLength(30)>
        <Column("LOWER_BODY", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property LowerBody() As String
        <MaxLength(30)>
        <Column("OFFICE_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property OfficeName() As String
        <MaxLength(60)>
        <Column("JOB_DESC", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property JobDescription() As String
        <MaxLength(60)>
        <Column("JOB_COMP_DESC", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property JobComponentDescription() As String
        <MaxLength(100)>
        <Column("ALERT_USER", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertUser() As String
        <Column("CP_ALERT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ClientPortalAlert() As Nullable(Of Short)
        <Column("NEW_ALERT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property NewAlert() As Nullable(Of Short)
        <Column("READ_ALERT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ReadAlert() As Nullable(Of Short)
        <MaxLength(6)>
        <Column("VN_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property VendorCode() As String
        <Column("DUE_DATE")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property DueDate() As Nullable(Of Date)
        <MaxLength(10)>
        <Column("TIME_DUE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property TimeDue() As String
        <Column("ALERT_STATE_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertStateID() As Nullable(Of Integer)
        <MaxLength(100)>
        <Column("ALERT_STATE_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertStateName() As String
        <Column("CURRENT_NOTIFY")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property CurrentNotify() As Nullable(Of Integer)
        <MaxLength(6)>
        <Column("CURRENT_NOTIFY_EMP_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property CurrentNotifyEmployeeCode() As String
        <MaxLength(64)>
        <Column("CURRENT_NOTIFY_EMP_FML", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property CurrentNotifyEmployeeFullName() As String
        <MaxLength(37)>
        <Column("GRP_OFFICE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupOffice() As String
        <MaxLength(49)>
        <Column("GRP_CLIENT", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupClient() As String
        <MaxLength(57)>
        <Column("GRP_DIVISION", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupDivision() As String
        <MaxLength(65)>
        <Column("GRP_PRODUCT", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupProduct() As String
        <MaxLength(69)>
        <Column("GRP_JOB", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupJob() As String
        <MaxLength(136)>
        <Column("GRP_COMPONENT", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupJobComponent() As String
        <Column("TASK_SEQ_NBR")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property TaskSequenceNumber() As Nullable(Of Short)
        <MaxLength(10)>
        <Column("TASK_FNC_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property TaskFunctionCode() As String
        <MaxLength(40)>
        <Column("TASK_FNC_DESC", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property TaskFunctionDescription() As String
        <Column("ESTIMATE_NUMBER")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property EstimateNumber() As Nullable(Of Integer)
        <Column("EST_COMPONENT_NBR")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property EstimateComponentNumber() As Nullable(Of Short)
        <MaxLength(67)>
        <Column("GRP_ESTIMATE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupEstimateNumber() As String
        <MaxLength(134)>
        <Column("GRP_ESTIMATE_COMPONENT", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupEstimateComponent() As String
        <MaxLength(51)>
        <Column("GRP_TASK", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupTask() As String
        <Required>
        <MaxLength(69)>
        <Column("GRP_CAMPAIGN", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupCampaign() As String
        <Column("CMP_IDENTIFIER")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property CampaignID() As Nullable(Of Integer)
        <MaxLength(10)>
        <Column("VER", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Version() As String
        <MaxLength(10)>
        <Column("BUILD", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Build() As String
        <Column("ALERT_SEQ_NBR")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertSequenceNumber() As Nullable(Of Short)
        <Required>
        <MaxLength(6)>
        <Column("GRP_PRIORITY", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupPriority() As String
        <MaxLength(64)>
        <Column("GRP_DUE_DATE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupDueDate() As String
        <MaxLength(64)>
        <Column("GRP_DUE_DATE_DISPLAY", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property GroupDueDateDisplay() As String
        <Column("ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property ID() As Nullable(Of Integer)
        <Column("SENT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Sent() As Nullable(Of Date)
        <System.ComponentModel.DataAnnotations.Schema.DatabaseGeneratedAttribute(ComponentModel.DataAnnotations.Schema.DatabaseGeneratedOption.Identity)>
        <Column("ALRT_NOTIFY_HDR_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertNotifyHeaderID() As Nullable(Of Integer)
        <MaxLength(100)>
        <Column("ALERT_NOTIFY_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertNotifyName() As String
        <Required>
        <MaxLength(1)>
        <Column("VN_NAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property VendorName() As String


#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.AlertID.ToString

        End Function

#End Region

    End Class

End Namespace

File: AdvantageFramework.Core/Database/Views/AlertView.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Views
{
    public partial class AlertView
    {
        [Column("ALERT_ID")]
        public int AlertId { get; set; }
        [Column("SUBJECT")]
        [StringLength(254)]
        public string Subject { get; set; }
        [Column("GENERATED", TypeName = "smalldatetime")]
        public DateTime? Generated { get; set; }
        [Column("PROCESSED", TypeName = "smalldatetime")]
        public DateTime? Processed { get; set; }
        [Column("USER_NAME")]
        [StringLength(64)]
        public string UserName { get; set; }
        [Column("EMP_CODE")]
        [StringLength(6)]
        public string EmpCode { get; set; }
        [Column("PRIORITY")]
        public int Priority { get; set; }
        [Required]
        [Column("TYPE")]
        [StringLength(40)]
        public string Type { get; set; }
        [Required]
        [Column("CATEGORY")]
        [StringLength(40)]
        public string Category { get; set; }
        [Column("ALERT_TYPE_ID")]
        public int AlertTypeId { get; set; }
        [Column("ALERT_CAT_ID")]
        public int AlertCatId { get; set; }
        [Column("ATTACHMENTCOUNT")]
        public int? Attachmentcount { get; set; }
        [Column("ALERT_LEVEL")]
        [StringLength(50)]
        public string AlertLevel { get; set; }
        [Column("OFFICE_CODE")]
        [StringLength(4)]
        public string OfficeCode { get; set; }
        [Column("CL_CODE")]
        [StringLength(6)]
        public string ClCode { get; set; }
        [Column("DIV_CODE")]
        [StringLength(6)]
        public string DivCode { get; set; }
        [Column("PRD_CODE")]
        [StringLength(6)]
        public string PrdCode { get; set; }
        [Column("CMP_CODE")]
        [StringLength(6)]
        public string CmpCode { get; set; }
        [Column("JOB_NUMBER")]
        public int? JobNumber { get; set; }
        [Column("JOB_COMPONENT_NBR")]
        public short? JobComponentNbr { get; set; }
        [Column("SENDER_EMP_CODE")]
        [StringLength(6)]
        public string SenderEmpCode { get; set; }
        [Column("DISMISSED_DATE")]
        [StringLength(10)]
        public string DismissedDate { get; set; }
        [Column("CL_NAME")]
        [StringLength(40)]
        public string ClName { get; set; }
        [Column("DIV_NAME")]
        [StringLength(40)]
        public string DivName { get; set; }
        [Column("PRD_DESCRIPTION")]
        [StringLength(40)]
        public string PrdDescription { get; set; }
        [Column("CMP_NAME")]
        [StringLength(60)]
        public string CmpName { get; set; }
        [Column("BODY", TypeName = "text")]
        public string Body { get; set; }
        [Column("LOWER_SUBJECT")]
        [StringLength(254)]
        public string LowerSubject { get; set; }
        [Column("LOWER_BODY")]
        [StringLength(30)]
        public string LowerBody { get; set; }
        [Column("OFFICE_NAME")]
        [StringLength(30)]
        public string OfficeName { get; set; }
        [Column("JOB_DESC")]
        [StringLength(60)]
        public string JobDesc { get; set; }
        [Column("JOB_COMP_DESC")]
        [StringLength(60)]
        public string JobCompDesc { get; set; }
        [Column("ALERT_USER")]
        [StringLength(100)]
        public string AlertUser { get; set; }
        [Column("CP_ALERT")]
        public short? CpAlert { get; set; }
        [Column("NEW_ALERT")]
        public short? NewAlert { get; set; }
        [Column("READ_ALERT")]
        public short? ReadAlert { get; set; }
        [Column("VN_CODE")]
        [StringLength(6)]
        public string VnCode { get; set; }
        [Column("DUE_DATE", TypeName = "smalldatetime")]
        public DateTime? DueDate { get; set; }
        [Column("TIME_DUE")]
        [StringLength(10)]
        public string TimeDue { get; set; }
        [Column("ALERT_STATE_ID")]
        public int? AlertStateId { get; set; }
        [Column("ALERT_STATE_NAME")]
        [StringLength(100)]
        public string AlertStateName { get; set; }
        [Column("CURRENT_NOTIFY")]
        public int? CurrentNotify { get; set; }
        [Column("CURRENT_NOTIFY_EMP_CODE")]
        [StringLength(6)]
        public string CurrentNotifyEmpCode { get; set; }
        [Column("CURRENT_NOTIFY_EMP_FML")]
        [StringLength(64)]
        public string CurrentNotifyEmpFml { get; set; }
        [Column("GRP_OFFICE")]
        [StringLength(37)]
        public string GrpOffice { get; set; }
        [Column("GRP_CLIENT")]
        [StringLength(49)]
        public string GrpClient { get; set; }
        [Column("GRP_DIVISION")]
        [StringLength(57)]
        public string GrpDivision { get; set; }
        [Column("GRP_PRODUCT")]
        [StringLength(65)]
        public string GrpProduct { get; set; }
        [Column("GRP_JOB")]
        [StringLength(69)]
        public string GrpJob { get; set; }
        [Column("GRP_COMPONENT")]
        [StringLength(136)]
        public string GrpComponent { get; set; }
        [Column("TASK_SEQ_NBR")]
        public short? TaskSeqNbr { get; set; }
        [Column("TASK_FNC_CODE")]
        [StringLength(10)]
        public string TaskFncCode { get; set; }
        [Column("TASK_FNC_DESC")]
        [StringLength(40)]
        public string TaskFncDesc { get; set; }
        [Column("ESTIMATE_NUMBER")]
        public int? EstimateNumber { get; set; }
        [Column("EST_COMPONENT_NBR")]
        public short? EstComponentNbr { get; set; }
        [Column("GRP_ESTIMATE")]
        [StringLength(67)]
        public string GrpEstimate { get; set; }
        [Column("GRP_ESTIMATE_COMPONENT")]
        [StringLength(134)]
        public string GrpEstimateComponent { get; set; }
        [Column("GRP_TASK")]
        [StringLength(51)]
        public string GrpTask { get; set; }
        [Required]
        [Column("GRP_CAMPAIGN")]
        [StringLength(69)]
        public string GrpCampaign { get; set; }
        [Column("CMP_IDENTIFIER")]
        public int? CmpIdentifier { get; set; }
        [Column("VERSION_ID")]
        [StringLength(10)]
        public string VersionId { get; set; }
        [Column("BUILD_ID")]
        [StringLength(10)]
        public string BuildId { get; set; }
        [Column("VER")]
        [StringLength(10)]
        public string Ver { get; set; }
        [Column("BUILD")]
        [StringLength(10)]
        public string Build { get; set; }
        [Column("ALERT_SEQ_NBR")]
        public short? AlertSeqNbr { get; set; }
        [Required]
        [Column("GRP_PRIORITY")]
        [StringLength(6)]
        public string GrpPriority { get; set; }
        [Column("GRP_DUE_DATE")]
        [StringLength(64)]
        public string GrpDueDate { get; set; }
        [Column("GRP_DUE_DATE_DISPLAY")]
        [StringLength(64)]
        public string GrpDueDateDisplay { get; set; }
        [Column("ID")]
        public int? Id { get; set; }
        [Column("SENT", TypeName = "smalldatetime")]
        public DateTime? Sent { get; set; }
        [Column("ALRT_NOTIFY_HDR_ID")]
        public int? AlrtNotifyHdrId { get; set; }
        [Column("ALERT_NOTIFY_NAME")]
        [StringLength(100)]
        public string AlertNotifyName { get; set; }
        [Required]
        [Column("VN_NAME")]
        [StringLength(1)]
        public string VnName { get; set; }
    }
}
